nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/pom.xml

```diff
@@ -159,62 +166,141 @@
             <artifactId>nifi-ssl-context-service</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.maven</groupId>
+            <artifactId>maven-artifact</artifactId>
+            <version>3.6.3</version>
+            <scope>test</scope>
+        </dependency>
         <dependency>
             <groupId>org.codehaus.groovy</groupId>
             <artifactId>groovy-json</artifactId>
             <version>${nifi.groovy.version}</version>
             <scope>test</scope>
-	</dependency>
-	<dependency>
+        </dependency>
+        <dependency>
             <groupId>org.elasticsearch.client</groupId>
             <artifactId>elasticsearch-rest-client</artifactId>
             <version>5.6.16</version>
         </dependency>
     </dependencies>
 
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-failsafe-plugin</artifactId>
+                    <version>3.0.0-M3</version>
+                </plugin>
+                <plugin>
+                    <groupId>com.github.alexcojocaru</groupId>
+                    <artifactId>elasticsearch-maven-plugin</artifactId>
+                    <version>6.19</version>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+
     <profiles>
         <profile>
+            <!-- use with integration-tests only (-default can be used if x-pack-ml permissions fixed) -->
             <id>integration-6</id>
```

Comment:
I run:

```bash
mvnd -P integration-tests clean verify
sleep 2s
mvnd -P integration-tests,integration-6 clean verify
sleep 2s
mvnd -P integration-tests,integration-7 clean verify
```

for:
* ES 5
* ES-6
* ES-7 (includes x-pack with the `-default` ES artefact for testing the "Point in Time" functionality)

Note that you need to `clean` for each run so that the Elasticsearch instance directory is deleted by Maven, otherwise it gets itself in a knot because of mixed libraries, etc. (e.g. trying to start an ES6 instance with ES5 libraries still in the `target` directory)

I'll add something into a README of the `pom.xml` to help future developers run the same

Reply:
Actually, I've just found that the `default` flavour of elasticsearch 7 wasn't running the tests (it spins up Elasticsearch but then doesn't run the `integration-tests`) so I've changed this to:
```bash
mvnd -P integration-tests,elasticsearch-oss clean verify
sleep 2s
mvnd -P integration-tests,elasticsearch-oss,elasticsearch-6 clean verify
sleep 2s
mvnd -P integration-tests,,elasticsearch-oss,integration-7 clean verify
sleep 2s
mvnd -P integration-tests,,elasticsearch-default,integration-7 clean verify
```

added in running ES 7 without X-Pack.

Updated the new README to include these